### DISCUSSION TITLE
Specify different name for the s3 bucket and the DNS domain of the website published with public-static-website module

### DISCUSSION THREAD

**Discussion by MicheledG** [gruntwork-io/knowledge-base#646] [Help] [answered]
Hi all 👋 

Is it possible to create a SPA website with an hosting S3 bucket name other than the website domain? 

I'm asking since the [public-static-website](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/modules/services/public-static-website) module does not expose this functionality, so the maximum length of a domain is equal to `63 chars - len("-cloudfront-logs") = 47 chars`.

Thx! 🙇 

---

<ins datetime="2023-01-26T13:41:41Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109837">Tracked in ticket #109837</a></p>
</ins>

**Comment by Etiene** [comment, marked as answer]
Hi! No, at the moment it is not possible as it would need to be exposed from module [`s3-static-website`](https://github.com/gruntwork-io/terraform-aws-static-assets/tree/main/modules/s3-static-website) first. Are you currently blocked by this? Is your domain name longer than 47 characters? If that's the case, we should start by filing a bug report at the [terraform-aws-static-assets](https://github.com/gruntwork-io/terraform-aws-static-assets) repo.

**Reply by MicheledG** [reply]
Hi @Etiene, thank you for the clarification!

I confirm that we have some use cases requiring long domain names (longer than 47 characters).

Should I proceed opening a bug on [terraform-aws-static-assets](https://github.com/gruntwork-io/terraform-aws-static-assets) repo to request this feature?

**Reply by Etiene** [reply]
@MicheledG yes, please! thanks
